### PR TITLE
Replace fromBinary() with deserialize() to make it consistent

### PR DESCRIPTION
**Pull request by TrustHenry**
We need serialize and deserialize to add a publickey in the block structure to be added.
issue #205

### CODE REVIEW

**Review by linked0 — APPROVED** [review]
LGTM!
It's needed for the issue #406 .

**Review by AndrejMitrovic — APPROVED** [review]

**Comment by AndrejMitrovic** [comment]
Feel free to merge when it passes the test-suite.

**Comment by AndrejMitrovic** [comment]
I think the `fromBinary` method should probably be removed? I don't see it used anywhere, and it looks like it should have been called `deserialize`.

**Comment by TrustHenry** [comment]
I agree with you
We can replace the `fromBinary` with `deserialize`.
I'll change it.

File: source/agora/common/crypto/Key.d [hunk 1]
```diff
@@ -198,20 +198,17 @@ public struct PublicKey
...
-            `PublicKey` Public key address
-
     ***************************************************************************/
 
-    public static PublicKey fromBinary (scope DeserializeDg dg) @safe
+    public void deserialize (scope DeserializeDg dg) @safe
     {
         alias DType = typeof(this.data);
-        return PublicKey(DType(dg(DType.sizeof)));
+        this.data = PublicKey(DType(dg(DType.sizeof)));
```
Comment (AndrejMitrovic) [inline source/agora/common/crypto/Key.d]:
```suggestion
        this.data = DType(dg(DType.sizeof));
```

Comment (TrustHenry) [reply source/agora/common/crypto/Key.d]: applied the suggested.

**Comment by Geod24** [comment]
> I think the fromBinary method should probably be removed?

Actually, it's something I wanted to experiment more with. Because `deserialize` is really problematic: some types make no sense default-initialized (`Point`), some types could be in an invalid state (`invariant` failing), etc... So initializing a type straight from binary, be it via a ctor or a specially-named method, is IMO a better road.
I went with a specially-named method because that's where you put the validation (as opposed to constructors).

**Pull request merged**